Controller/Process/Json.php

```diff
@@ -204,6 +213,15 @@ protected function _processNotification($response, $notificationMode)
                 );
                 return false;
             }
+
+            if ($this->hmacSignature->isHmacSupportedEventCode($response)) {
+                //Validate the Hmac calculation
+                if (!$this->hmacSignature->isValidNotificationHMAC($this->configHelper->getNotificationsHmacKey(),
```

Comment:
so if hmac key is not set then we should not valdiate the hmac signature.

Reply:
@AlexandrosMor still valid concern can you fix this ?

Reply:
ok ignore my concern. So if the NotificationsIpHmacCheck is enabled it should valdiate the HMAC at all times. The check if this is set is above so this is solved.